src/lib/ledger_builder_controller/transition_logic_state.ml

```diff
@@ -97,7 +115,18 @@ module Make (Inputs : Inputs.Base.S) :
   let apply_all t changes =
     assert_state_valid t ;
     let t' = List.fold changes ~init:t ~f:apply in
-    assert_state_valid t' ; t'
+    try assert_state_valid t' ; t' with exn ->
+      Printf.printf
```

Comment:
replace printfs with logger?

Reply:
I used printf here since this happens right before a crash and It was going to be more difficult to pull the rendered dot graph out from the sexp format. If you feel adamant that it should still be a logger call, I can lift it out and write it to a file or something.

Reply:
Yes logger call, we should avoid writing raw data to stdout